doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -1758,6 +1670,12 @@ Tactic notations allow to customize the syntax of tactics. They have the followi
    following table. The last command gives examples of tactics which use
    the corresponding kind of argument.
 
+   .. todo this table is too wide in HTML, I see a horizontal scroll bar.
+      Any ideas on easy ways to make it fit?  The hard way is
+
+   .. todo At the moment I splice uconstr.  Should I bring it back?
+      An untyped term is one that doesn't get typechecked??
```

Comment:
I don't think so. An untyped term just doesn't get typed *before* being passed as argument to the tactic. It avoids getting errors for holes whose type could not be inferred as in the following example:

```coq
Tactic Notation "refine_typed" constr(x) := refine x.
Tactic Notation "refine_untyped" uconstr(x) := refine x.

Goal True -> True.
Fail refine_typed (fun x => _).
refine_untyped (fun x => _).
```

However, the tactic itself will be able to type-check the term using additional information (for instance by matching the term with the current goal).

Reply:
OK, untyped is not a good description.  How about something like "term_with_holes"?

Reply:
I think untyped is fine.

Reply:
Yes, untyped is fine.  Note that in your particular example, `open_constr` would also work in place of `uconstr`.  You need sightly more exotic terms to see the difference between `uconstr` and `open_constr`

Reply:
Is `open_constr` missing from the table?

Reply:
> Is `open_constr` missing from the table?

Yes.  I will add a todo.

Reply:
Yes, apparently `open_constr` is missing.  Perhaps a good example is `open_constr` as in `epose`.  It was apparently also missing from the grammar list of `tactic_argument_type` (why did the grammar for `tactic_argument_type` disappear?)

Reply:
Is `tactic_argument_type` the right string?  I don't see that string in the source tree.

Looks like there are at least a dozen more that are not described including a bunch for SSR.  I don't want to add them right now in this PR (but maybe soon when we do the Ltac chapter).  I left this in the file:

```
   .. todo 'open_constr' appears to be another possible value based on the
      the message from "Tactic Notation open_constr := idtac".
      Also (at least) "ref", "string", "preident", "int" and "ssrpatternarg".
      (from reading .v files).
      Looks like any string passed to "make0" in the code is valid.  But do
      we want to support all these?
```

Reply:
https://github.com/coq/coq/blob/master/doc/sphinx/user-extensions/syntax-extensions.rst#tactic-notations
```rst
Tactic Notations
-----------------

Tactic notations allow to customize the syntax of tactics. They have the following syntax:

.. productionlist:: coq
   tacn                 : Tactic Notation [`tactic_level`] [`prod_item` … `prod_item`] := `tactic`.
   prod_item            : `string` | `tactic_argument_type`(`ident`)
   tactic_level         : (at level `num`)
   tactic_argument_type : ident | simple_intropattern | reference
                        : hyp | hyp_list | ne_hyp_list
                        : constr | uconstr | constr_list | ne_constr_list
                        : integer | integer_list | ne_integer_list
                        : int_or_var | int_or_var_list | ne_int_or_var_list
                        : tactic | tactic0 | tactic1 | tactic2 | tactic3
                        : tactic4 | tactic5
```
This seems to have been deleted in the current version of the file

Reply:
Well, everything listed for `tactic_argument_type` is in the table in the `Tactic Notation` command.  The doc for the command says "ident is the name of a grammar nonterminal listed in the table below."  The table also makes clear the list variants apply to all the other entries.  You want to have both the table AND the grammar?

Reply:
No, what you did is reasonable. From a syntax point of view, this is just an ident. We just have to document that this ident must be taken in this list. If you give an incorrect ident, it is not a parsing error but an interpretation error.

Reply:
Is this conversation resolved now?  I can't tell.  (It seems like it might be to me.)

Reply:
The part about deleting `tactic_argument_type` is resolved since we agree.  I still wonder whether it makes sense to bring back `uconstr` in the grammar to make things more explicit.  I updated the todo to reflect this.

If I reply to a comment, I let the reviewer resolve the item.  Sometimes I leave comments unresolved for a while while I think about them some more.  The difference is communicated by telepathy.

A binary `Resolved` state doesn't work as well when multiple reviewers weigh in on a single comment.  May need to allow more time so both reviewers can see the most recent comments.  Resolved comments are harder to find in GitHub than unresolved comments..